x-pack/solutions/security/plugins/elastic_assistant/server/routes/attack_discovery/helpers/generate_and_update_discoveries.ts

```diff
@@ -54,7 +56,12 @@ export const generateAndUpdateAttackDiscoveries = async ({
   savedObjectsClient,
   telemetry,
   withReplacements,
-}: GenerateAndUpdateAttackDiscoveriesParams) => {
+}: GenerateAndUpdateAttackDiscoveriesParams): Promise<{
+  anonymizedAlerts?: Document[];
+  attackDiscoveries?: AttackDiscoveryApiAlert[];
+  error?: { message?: string }; // for compatibility with legacy internal API error handling
```

Comment:
When the legacy `/internal` APIs are deleted (in a future PR) this function should reject instead of returning an optional `error`, and most of the other properties should be non-optional